### ISSUE TITLE
Add resource requirements for Workspace statefulsets

### ISSUE THREAD

**Issue by Vafilor** [onepanelio/core#465]
# Background

When we create a Workspace, it is stored as a Statefulset. We use a nodeSelector to target a machine. However, we don't specify the resource requirements such as cpu/memory/gpu. This means that the scheduler could put multiple pods onto the same node, so we're not getting the full machine capacity for our Workspace.

In order to get the full machine capacity, we need to request the full machine resources available.

# Task

The user can specify the resources in the configuration provided in the `params.yaml`.
The system needs to read this, and if it is set, add it to the statefulset manifest when creating a workspace.

We already do this for workflows, here is the code 

https://github.com/onepanelio/core/pull/393/files#diff-2df54c8de22b7ec30c64350c5cbfd591R147

# Testing
- [ ] configure params.yaml with resource requirements and make sure it is saved correctly in the final manifest
- [ ] create a workspace and make sure the manifest adds the resource requirements
- [ ] repeat the above, but use a different node pool to make sure the resource requirements are changing appropriately

**Comment by aleksandrmelnikov** [comment]
An update on this.

Through recent discussions about this functionality, we want to increase the automation regarding the nodes available and their resources.
For this ticket specifically, we're dealing with node resources and making sure the pod can use all of the resources without having to share.

For example, we want CVAT workspace to be able to use the full node resources, such as GPU and CPU, without sharing with another CVAT workspace.

We explored this solution:
Add an extra container to a given workspace.
The container is minimal, such as alpine:linux, and instructed to sleep forever.
- So the container takes almost no resources to run
- Has to run forever so argo doesn't think the workflow is crashing

The extra container has extra fields to request resources, such as `cpu: 4000m` (4 cpus) and `nvidia|amd.com:1` to request GPUs.

This generally works.
We can read the users configuration values from params.yaml and request them when the workspace is created, for a selected node.

To improve automation, we wanted to figure out this capacity dynamically, without the user setting the information in params.yaml.

So, we added code to look up the running nodes and retrieve their available capacity for allocation.
- This also worked.
- Some issues with figuring out the exact amount, because I had to tweak and set 90% of the values reported for the pod to schedule properly.

This solution had a big issue: we required the nodes to be running in order to see their capacity.
- Some nodepools would be at 0 and thus scale up when required.
There is no easy way to get this information from k8s.
More difficulty, the cloud providers (such as AKS) do not make their cluster-autoscaler available in the cluster.
- It gets used, but it can't be outright accessed

So finally, Rush discovered a solution of pod anti-affinity to schedule the pods.
- We don't worry about resources, because we can ensure only one pod per node. New pods will be forced to go to other nodes.
- The autoscaler will still schedule new nodes as needed.
